src/backend/d3d12/ShaderModuleD3D12.cpp

```diff
@@ -24,15 +24,54 @@ namespace d3d12 {
         spirv_cross::CompilerHLSL compiler(builder->AcquireSpirv());
 
         spirv_cross::CompilerHLSL::Options options;
-        options.shader_model = 40;
+        options.shader_model = 51;
         options.flip_vert_y = false;
         options.fixup_clipspace = true;
 
         compiler.set_options(options);
 
         ExtractSpirvInfo(compiler);
 
+        enum RegisterType {
+            Buffer,
+            UnorderedAccess,
+            Texture,
+            Sampler,
+            Count,
+        };
+
+        std::array<uint32_t, RegisterType::Count * kMaxBindGroups> baseRegisters = {};
```

Comment:
nit: this could be an std::array<uint32_t, kMaxBindGroups> on the stack in the lambda, removing the need for the offset argument.